Projects/UOContent/Mobiles/AI/BaseAI.cs

```diff
@@ -1276,44 +1075,56 @@ public virtual bool DoOrderCome()
             return true;
         }
 
-        var iCurrDist = (int)m_Mobile.GetDistanceToSqrt(m_Mobile.ControlMaster);
+        var currentDistance = (int)m_Mobile.GetDistanceToSqrt(m_Mobile.ControlMaster);
 
-        if (iCurrDist > m_Mobile.RangePerception)
+        if (currentDistance > m_Mobile.RangePerception)
         {
-            if (m_Mobile.Debug)
-            {
-                m_Mobile.DebugSay("I have lost my master. I stay here");
-            }
-
-            m_Mobile.ControlTarget = null;
-            m_Mobile.ControlOrder = OrderType.None;
+            HandleLostMaster();
         }
         else
         {
-            if (m_Mobile.Debug)
-            {
-                m_Mobile.DebugSay("My master told me come");
-            }
+            HandleComeOrder(currentDistance);
+        }
 
-            // Not exactly OSI style, but better than nothing.
-            var bRun = iCurrDist > 5;
+        return true;
+    }
 
-            if (WalkMobileRange(m_Mobile.ControlMaster, 1, bRun, 0, 1))
-            {
-                if (m_Mobile.Combatant?.Deleted == false && m_Mobile.Combatant.Alive &&
-                    !m_Mobile.Combatant.IsDeadBondedPet)
-                {
-                    m_Mobile.Warmode = true;
-                    // m_Mobile.Direction = m_Mobile.GetDirectionTo(m_Mobile.Combatant, bRun);
-                }
-                else
-                {
-                    m_Mobile.Warmode = false;
-                }
-            }
+    private void HandleLostMaster()
+    {
+        if (m_Mobile.Debug)
+        {
+            m_Mobile.DebugSay("Master is missing. Staying put.");
         }
+    
+        m_Mobile.ControlTarget = null;
+        m_Mobile.ControlOrder = OrderType.None;
+    }
 
-        return true;
+    private void HandleComeOrder(int currentDistance)
+    {
+        if (m_Mobile.Debug)
+        {
+            m_Mobile.DebugSay("I am ordered to come.");
+        }
+    
+        var shouldRun = currentDistance > 5;
+    
+        if (WalkMobileRange(m_Mobile.ControlMaster, 1, shouldRun, 0, 1))
+        {
+            UpdateCombatantState();
+        }
+    }
+
+    private void RefreshCombatantState()
+    {
+        if (IsValidCombatant(m_Mobile.Combatant))
+        {
+            m_Mobile.Warmode = true;
+        }
+        else
+        {
+            m_Mobile.Warmode = false;
+        }
```

Comment:
Can be compressed to this:

```suggestion
        m_Mobile.Warmode = IsValidCombatant(m_Mobile.Combatant);
```